diff --git a/swig/python/CMakeLists.txt b/swig/python/CMakeLists.txt
index 74c801b47a..f44ecc758b 100644
--- a/swig/python/CMakeLists.txt
+++ b/swig/python/CMakeLists.txt
@@ -209,16 +209,11 @@ mark_as_advanced(CASADI_PYTHON_PIP_METADATA_INSTALL)
 set(CASADI_PYTHON_PIP_METADATA_INSTALLER "cmake" CACHE STRING "Specify the string to identify the pip Installer. Default: cmake, change this if you are using another tool.")
 mark_as_advanced(CASADI_PYTHON_PIP_METADATA_INSTALLER)
 if (CASADI_PYTHON_PIP_METADATA_INSTALL)
-  if (WIN32)
-    set(NEW_LINE "\n\r")
-  else()
-    set(NEW_LINE "\n")
-  endif()
   file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/METADATA "")
-  file(APPEND ${CMAKE_CURRENT_BINARY_DIR}/METADATA "Metadata-Version: 2.1${NEW_LINE}")
-  file(APPEND ${CMAKE_CURRENT_BINARY_DIR}/METADATA "Name: casadi${NEW_LINE}")
-  file(APPEND ${CMAKE_CURRENT_BINARY_DIR}/METADATA "Version: ${CASADI_VERSION}${NEW_LINE}")
-  file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/INSTALLER "${CASADI_PYTHON_PIP_METADATA_INSTALLER}${NEW_LINE}")
+  file(APPEND ${CMAKE_CURRENT_BINARY_DIR}/METADATA "Metadata-Version: 2.1\n")
+  file(APPEND ${CMAKE_CURRENT_BINARY_DIR}/METADATA "Name: casadi\n")
+  file(APPEND ${CMAKE_CURRENT_BINARY_DIR}/METADATA "Version: ${CASADI_VERSION}\n")
+  file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/INSTALLER "${CASADI_PYTHON_PIP_METADATA_INSTALLER}\n")
   install(
     FILES "${CMAKE_CURRENT_BINARY_DIR}/METADATA" "${CMAKE_CURRENT_BINARY_DIR}/INSTALLER"
     DESTINATION ${PYTHON_PREFIX}/casadi-${CASADI_VERSION}.dist-info
