diff --git a/casadi/interfaces/qpoases/qpoases_interface.hpp b/casadi/interfaces/qpoases/qpoases_interface.hpp
index b4d8ea9e52..644ffa52d9 100644
--- a/casadi/interfaces/qpoases/qpoases_interface.hpp
+++ b/casadi/interfaces/qpoases/qpoases_interface.hpp
@@ -31,6 +31,8 @@
 #include <casadi/interfaces/qpoases/casadi_conic_qpoases_export.h>
 #include <qpOASES.hpp>
 
+namespace qpOASES = casadi_qpOASES; 
+
 /** \defgroup plugin_Conic_qpoases Title
     \par
 
diff --git a/external_packages/qpOASES/include/qpOASES/Types.hpp b/external_packages/qpOASES/include/qpOASES/Types.hpp
index e9d06439b3..a309e25927 100644
--- a/external_packages/qpOASES/include/qpOASES/Types.hpp
+++ b/external_packages/qpOASES/include/qpOASES/Types.hpp
@@ -104,16 +104,16 @@
 #else
 
     /** Macro for switching on/off the beginning of the qpOASES namespace definition. */
-    #define BEGIN_NAMESPACE_QPOASES  namespace qpOASES {
+    #define BEGIN_NAMESPACE_QPOASES  namespace casadi_qpOASES {
 
     /** Macro for switching on/off the end of the qpOASES namespace definition. */
     #define END_NAMESPACE_QPOASES    }
 
     /** Macro for switching on/off the use of the qpOASES namespace. */
-    #define USING_NAMESPACE_QPOASES  using namespace qpOASES;
+    #define USING_NAMESPACE_QPOASES  using namespace casadi_qpOASES;
 
     /** Macro for switching on/off references to the qpOASES namespace. */
-    #define REFER_NAMESPACE_QPOASES  qpOASES::
+    #define REFER_NAMESPACE_QPOASES  casadi_qpOASES::
 
 #endif
 
