diff --git a/CMakeLists.txt b/CMakeLists.txt
index 67696e5823..54bb09bd95 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -870,6 +870,12 @@ foreach(PKG HIGHS PROXQP OSQP SUPERSCS BONMIN IPOPT CBC CLP MUMPS SPRAL HSL METI
       get_property(PACKAGES_NOT_FOUND_BACKUP GLOBAL PROPERTY PACKAGES_NOT_FOUND)
       get_property(PACKAGES_FOUND_BACKUP GLOBAL PROPERTY PACKAGES_FOUND)
       find_package(${PKG})
+      if(PKG STREQUAL "PROXQP")
+        find_package(proxsuite)
+	      set(PROXQP_FOUND ${proxsuite_FOUND})
+      else()
+        find_package(${PKG})
+      endif()
     endif()
     if (WITH_BUILD_${PKG} OR (WITH_BUILD_REQUIRED AND NOT ${PKG}_FOUND))
 
diff --git a/casadi/interfaces/proxqp/CMakeLists.txt b/casadi/interfaces/proxqp/CMakeLists.txt
index 72211cb757..ef750a64ad 100644
--- a/casadi/interfaces/proxqp/CMakeLists.txt
+++ b/casadi/interfaces/proxqp/CMakeLists.txt
@@ -7,8 +7,11 @@ casadi_plugin(Conic proxqp
   proxqp_interface.cpp
   proxqp_interface_meta.cpp)
 
-
-casadi_plugin_link_libraries(Conic proxqp proxqp)
+if(WITH_BUILD_PROXQP)
+  casadi_plugin_link_libraries(Conic proxqp proxqp)
+else()
+  casadi_plugin_link_libraries(Conic proxqp proxsuite::proxsuite)
+endif()
 
 if("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang" OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
   set_target_properties(casadi_conic_proxqp PROPERTIES COMPILE_FLAGS "-Wno-unused-variable -Wno-unknown-warning-option -Wno-type-limits")
